src/common/entity/compute_attribute_display.ts

```diff
@@ -0,0 +1,52 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import { EntityRegistryEntry } from "../../data/entity_registry";
+import { HomeAssistant } from "../../types";
+import { LocalizeFunc } from "../translations/localize";
+import { computeDomain } from "./compute_domain";
+
+export const computeAttributeValueDisplay = (
+  localize: LocalizeFunc,
+  stateObj: HassEntity,
+  entities: HomeAssistant["entities"],
+  attribute: string,
+  value?: any
+): string => {
+  const entityId = stateObj.entity_id;
+  const attributeValue =
+    value !== undefined ? value : stateObj.attributes[attribute];
+  const domain = computeDomain(entityId);
+  const entity = entities[entityId] as EntityRegistryEntry | undefined;
+  const translationKey = entity?.translation_key;
+
+  return (
+    (translationKey &&
+      localize(
+        `component.${entity.platform}.entity.${domain}.${translationKey}.state_attributes.${attribute}.state.${attributeValue}`
+      )) ||
```

Comment:
Is there an easy way to not use the `||` here and be more explicit about when each key should apply?

I don't expect to make the core keys type checked anytime soon, but this style of coding `localize` with the expectation it will return `""` if the key doesn't exist is prone to error, makes checking the key impossible, and is just hard to read and understand.

Reply:
Every components and custom components can provide translation. We check component translations, the domain translations and then we fallback to the attribute key. It's just fallback rules. 
As attributes translations are now managed by the back-end, I don't known if we can check anything in the front-end part 😞.

Reply:
We don't even check it at the backend, as a translation for these is actually optional.
The value itself doesn't have to be translatable.